Java 20 date time formatting

jamesfredley opened the pull request:
In Java 20+, [Unicode CLDR42](https://cldr.unicode.org/downloads/cldr-42 ) was implemented which changed the space character preceding the period (AM or PM) in formatted date/time text from a standard space (" ") to a narrow non-breaking space (NNBSP: "\u202F"). Additionally, when using the LONG or FULL timeStyle with dateStyle, the date and time separator has changed from ' at ' to ', '. IE. January 5, 1941, 8:00:00 AM UTC vs. January 5, 1941 at 8:00:00 AM UTC

We already had Java 8 and 17 specific tests.  For each of the Java version specific scenarios covered by prior tests, a Java 20+ test was added.

jamesfredley committed:
Update tests for Java 20+

jamesfredley committed:
Update formatDate documentation with Java 20+ changes

jdaugherty reviewed (approved):
This change needs called out in the v6 to 7 upgrade notes too

matrei reviewed (approved):
Would using a Unicode escape sequence be preferable to improve readability, rather than including a non-visible character in the comparison strings?

jamesfredley committed:
Use Unicode escape sequence instead of non-visible character

jamesfredley committed:
Add Date Format changes in Java 20+ to v6 to 7 upgrade notes

jamesfredley commented:
I have updated to tests to use \u202F instead of NNBSP.  IntelliJ makes is easy to see, but it is invisible in the diff.

Also added details to Breaking changes section of v6 to 7 upgrade notes.
